### PR TITLE
Adding more sql functions year, day, hour, minute, second

### PR DESCRIPTION
**Pull request by hichem-fazai**
This PR is for adding more functions that extracts a date unit from a datetime column like year, day, hour, minute and second.

### CODE REVIEW

**Comment by hichem-fazai** [comment]
I will work on more functions that works with strings, integers,.. etc.
And then create tests.

File: exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/JavaDateFunctions.kt [hunk 1]
```diff
@@ -25,12 +25,27 @@ class CurrentDateTime : Function<LocalDateTime>(JavaLocalDateTimeColumnType.INST
...
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) = queryBuilder {
+        when (currentDialect) {
+            is PostgreSQLDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is OracleDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is MysqlDialect -> append("EXTRACT(YEAR FROM ", expr, ")")
+            is SQLServerDialect -> append("YEAR(", expr, ")")
+            is MariaDBDialect -> append("YEAR(", expr, ")")
+            is SQLiteDialect -> append("STRFTIME('%Y',", expr, ")")
+            is H2Dialect -> append("YEAR(", expr, ")")
+            else -> append("YEAR(", expr, ")")
```
Comment (hichem-fazai) [inline exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/JavaDateFunctions.kt]: In the else statement i thought of two solutions:
- To throw an Exception telling that the current database dialect is not supprted. So if a new dialect is added we must add it's syntax here.
- Return Function-name(column-name) which can be a "default" syntax if a new dialect is added to Exposed.
So what you think is best to do?

Comment (Tapac) [reply exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/JavaDateFunctions.kt]: @hichem-fazai , sorry I was on a vacation and missed your comment. 
I guess what `UnsupportedByDialectException` should be fine.

Comment (hichem-fazai) [reply exposed-java-time/src/main/kotlin/org/jetbrains/exposed/sql/JavaDateFunctions.kt]: Hi @Tapac, ok perfect :)
I added a commit to handle the exception for all the functions.

**Comment by Tapac** [comment]
Rejected in favor to #707

**Pull request closed without merging**